### PR TITLE
VULTR: adopt diff2 in full mode

### PR DESCRIPTION
**Pull request by androw**
Tried both
```
    go test -v -verbose -provider VULTR
```
and
```
    go test -v -verbose -provider VULTR -diff2
```
and both succeed

### CODE REVIEW

File: providers/vultr/vultrProvider.go [hunk 1]
```diff
@@ -134,47 +134,82 @@ func (api *vultrProvider) GetDomainCorrections(dc *models.DomainConfig) ([]*mode
...
+		for _, mod := range create {
+			r := toVultrRecord(dc, mod.Desired, "0")
+			corrections = append(corrections, &models.Correction{
+				Msg: mod.String(),
+				F: func() error {
+					_, err := api.client.DomainRecord.Create(context.Background(), dc.Name, &govultr.DomainRecordReq{Name: r.Name, Type: r.Type, Data: r.Data, TTL: r.TTL, Priority: &r.Priority})
+					return err
+				},
+			})
+		}
```
Comment (tlimoncelli) [inline providers/vultr/vultrProvider.go:174]: Add a `return corrections, nil` to this section then remove `else {`.  That way when the references to `pkg/diff` go away, the diff won't include a reformatted "else".

File: providers/vultr/vultrProvider.go [hunk 2]
```diff
@@ -134,47 +134,82 @@ func (api *vultrProvider) GetDomainCorrections(dc *models.DomainConfig) ([]*mode
...
+					Msg: change.Msgs[0],
+					F: func() error {
+						_, err := api.client.DomainRecord.Create(context.Background(), dc.Name, &govultr.DomainRecordReq{Name: r.Name, Type: r.Type, Data: r.Data, TTL: r.TTL, Priority: &r.Priority})
+						return err
+					},
+				})
+			case diff2.CHANGE:
+				r := toVultrRecord(dc, change.New[0], change.Old[0].Original.(govultr.DomainRecord).ID)
+				corrections = append(corrections, &models.Correction{
+					//Msg: change.MsgsJoined,
```
Comment (tlimoncelli) [inline providers/vultr/vultrProvider.go:196]: Remove commented out code

File: providers/vultr/vultrProvider.go [hunk 3]
```diff
@@ -134,47 +134,82 @@ func (api *vultrProvider) GetDomainCorrections(dc *models.DomainConfig) ([]*mode
...
+					Msg: fmt.Sprintf("%s; Vultr RecordID: %v", change.Msgs[0], r.ID),
+					F: func() error {
+						return api.client.DomainRecord.Update(context.Background(), dc.Name, r.ID, &govultr.DomainRecordReq{Name: r.Name, Type: r.Type, Data: r.Data, TTL: r.TTL, Priority: &r.Priority})
+					},
+				})
+			case diff2.DELETE:
+				id := change.Old[0].Original.(govultr.DomainRecord).ID
+				corrections = append(corrections, &models.Correction{
+					//Msg: change.MsgsJoined,
```
Comment (tlimoncelli) [inline providers/vultr/vultrProvider.go:205]: Remove commented out code

**Review by tlimoncelli — COMMENTED** [review]
Looking good!  Just a few minor suggestions.

File: providers/vultr/vultrProvider.go (hunk 1 above)
Comment (androw) [reply providers/vultr/vultrProvider.go:174]: Done, can you confirm it's okay like that?

**Review by tlimoncelli — APPROVED** [review]

**Comment by tlimoncelli** [comment]
P.S. Thanks for taking the time to implement a `diff2.By*()` call rather than using the backwards compatibility API!  I appreciate the extra effort.  I hope the new interface was easy to use and would appreciate feedback.

**Comment by androw** [comment]
Yup, it was easy to migrate just by looking at the example code and at the others providers implementation.

**Pull request merged**